contracts/modules/TransferManager/ManualApprovalTransferManager.sol

```diff
@@ -111,51 +101,233 @@ contract ManualApprovalTransferManager is ITransferManager {
+    * @param _allowances is the array of approved amounts 
+    * @param _expiryTimes is the array of the times until which eath transfer is allowed
+    * @param _descriptions is the description array for these manual approvals
+    */
+    function addManualApprovalMulti(
+        address[] _from,
+        address[] _to,
+        uint256[] _allowances,
+        uint256[] _expiryTimes,
+        bytes32[] _descriptions
+    ) 
+        external 
+        withPerm(TRANSFER_APPROVAL)
+    {
+        _checkInputLengthArray(_from, _to, _allowances, _expiryTimes, _descriptions);
+        for (uint256 i = 0; i < _from.length; i++){
+            _addManualApproval(_from[i], _to[i], _allowances[i], _expiryTimes[i], _descriptions[i]);
+        }
+    }
+
+    /**
+    * @notice Modify the existing manual approvals
+    * @param _from is the address from which transfers are approved
+    * @param _to is the address to which transfers are approved
+    * @param _expiryTime is the time until which the transfer is allowed
+    * @param _changedAllowance is the changed allowance
+    * @param _description Description about the manual approval
+    * @param _change uint values which tells whether the allowances will be increased (1) or decreased (0)
+    * or any value when there is no change in allowances
     */
-    function addManualBlocking(address _from, address _to, uint256 _expiryTime) public withPerm(TRANSFER_APPROVAL) {
+    function modifyManualApproval(
+        address _from,
+        address _to,
+        uint256 _expiryTime,
+        uint256 _changedAllowance,
+        bytes32 _description,
+        uint8 _change
+    ) 
+        external
+        withPerm(TRANSFER_APPROVAL)
+    {
+        _modifyManualApproval(_from, _to, _expiryTime, _changedAllowance, _description, _change);
+    }
+
+    function _modifyManualApproval(
+        address _from,
+        address _to,
+        uint256 _expiryTime,
+        uint256 _changedAllowance,
+        bytes32 _description,
+        uint8 _change
+    ) 
+        internal 
+    {
         require(_to != address(0), "Invalid to address");
         /*solium-disable-next-line security/no-block-members*/
         require(_expiryTime > now, "Invalid expiry time");
-        require(manualBlockings[_from][_to].expiryTime == 0, "Blocking already exists");
-        manualBlockings[_from][_to] = ManualBlocking(_expiryTime);
-        emit AddManualBlocking(_from, _to, _expiryTime, msg.sender);
+        require(manualApprovals[_from][_to].allowance != 0, "Approval does not exists");
+        uint256 currentAllowance = manualApprovals[_from][_to].allowance;
+        uint256 newAllowance;
+        if (_change == 1) {
+            // Allowance get increased
+            manualApprovals[_from][_to].allowance = (
+            manualApprovals[_from][_to].allowance.add(_changedAllowance));
+            newAllowance = currentAllowance.add(_changedAllowance);
```

Comment:
A minor optimization can be that we calculate `newAllowance` first and then set `manualApprovals[_from][_to].allowance = newAllowance`. The same optimization can also be done in the else if case below.